pkg/controller/groups/samlgrouplinks/controller.go

```diff
@@ -0,0 +1,189 @@
+	errGetFailed            = "cannot get Gitlab SamlGroupLink"
+	errCreateFailed         = "cannot create Gitlab SamlGroupLink"
+	errDeleteFailed         = "cannot delete Gitlab SamlGroupLink"
+	errSamlGroupLinktNotFound = "cannot find Gitlab SamlGroupLink"
+	errMissingGroupID       = "missing Spec.ForProvider.GroupID"
+)
+
+func SetupSamlGroupLink(mgr ctrl.Manager, o controller.Options) error {
+	name := managed.ControllerName(v1alpha1.SamlGroupLinkKind)
+	cps := []managed.ConnectionPublisher{managed.NewAPISecretPublisher(mgr.GetClient(), mgr.GetScheme())}
+
+	if o.Features.Enabled(features.EnableAlphaExternalSecretStores) {
+		cps = append(cps, connection.NewDetailsManager(mgr.GetClient(), secretstoreapi.StoreConfigGroupVersionKind))
+	}
+
+	reconcilerOpts := []managed.ReconcilerOption{
+		managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newGitlabClientFn: groups.NewSamlGroupLinkClient}),
+		managed.WithInitializers(),
+		managed.WithPollInterval(o.PollInterval),
+		managed.WithLogger(o.Logger.WithValues("controller", name)),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+		managed.WithConnectionPublishers(cps...),
+	}
+	if o.Features.Enabled(features.EnableAlphaManagementPolicies) {
+		reconcilerOpts = append(reconcilerOpts, managed.WithManagementPolicies())
+	}
+
+	r := managed.NewReconciler(mgr, 
+		resource.ManagedKind(v1alpha1.SamlGroupLinkGroupVersionKind),
+		reconcilerOpts...)
+	
+		return ctrl.NewControllerManagedBy(mgr).
+			Named(name).
+			For(&v1alpha1.SamlGroupLink{}).
+			Complete(r)
+}
+
+type connector struct {
+	kube client.Client
+	newGitlabClientFn func(cfg clients.Config) groups.SamlGroupLinkClient
+}
+
+func (c * connector)  Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.SamlGroupLink)
+	if !ok {
+		return nil, errors.New(errNotSamlGroupLink)
+	}
+
+	cfg, err := clients.GetConfig(ctx, c.kube, cr)
+	if err != nil {
+		return nil, err 
+	}
+	return &external{kube: c.kube, client: c.newGitlabClientFn(*cfg)}, nil
+}
+
+type external struct {
+	kube client.Client
+	client groups.SamlGroupLinkClient
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.SamlGroupLink)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotSamlGroupLink)
+	}
+
+	samlGroupName := *cr.Spec.ForProvider.Name
+	if samlGroupName == "" {
+		return managed.ExternalObservation{}, nil
+	}
+
+	if cr.Spec.ForProvider.GroupID == nil {
+		return managed.ExternalObservation{}, errors.New(errMissingGroupID)
+	}
+
+	groupLink, res, err := e.client.GetGroupSAMLLink(*cr.Spec.ForProvider.GroupID, samlGroupName)
+	if err != nil{
+		if clients.IsResponseNotFound(res) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, errors.Wrap(err, errGetFailed)
```

Comment:
Can be simplied:

```suggestion
		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(err, clients.IsResponseNotFound), errGetFailed)
```

Reply:
I tried to implement this simplification but it caused errors with the resources.Ignore method. The other controller files in the groups folder use the original method, is it necessary?

Reply:
@MisterMX any thought on this?

Reply:
Can you describe what is not working? [`resource.Ignore`](https://pkg.go.dev/github.com/crossplane/crossplane-runtime/pkg/resource#Ignore) is used in many providers without issues.

Reply:
The issue was the clients.IsResponseNotFound function does not implement the right parameter. expected was error not *gitlab.Response. I reworked the statement to have an IsErrorSamlGroupLinkNotFound function in clients that works. Modified the Unit tests and tested it in my running instance. 

``` golang
	if err != nil {
		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(groups.IsErrorSamlGroupLinkNotFound, err), errGetFailed)
	}
```